### PR TITLE
[data grid] Start extracting material slots to a separate directory

### PR DESCRIPTION
**Pull request by cherniavskii**
Extracted from https://github.com/mui/mui-x/pull/7095

### CODE REVIEW

**Review by siriwatknp — APPROVED** [review]
👍 Great start. Do we need another reviewer from MUI X team?

File: packages/grid/x-data-grid-pro/src/material/index.ts [hunk 1]
```diff
@@ -0,0 +1,15 @@
...
+import { GridProColumnMenu } from '../components/GridProColumnMenu';
+
+const iconSlots: GridProIconSlotsComponent = {
+  ColumnMenuPinRightIcon: GridPushPinRightIcon,
+  ColumnMenuPinLeftIcon: GridPushPinLeftIcon,
+};
+
+const materialSlots = {
+  ...iconSlots,
+  ColumnMenu: GridProColumnMenu,
```
Comment (MBilalShafi) [inline packages/grid/x-data-grid-pro/src/material/index.ts:12]: Should `ColumnMenu` be included in `material` slots for pro and premium packages keeping in mind it's a custom grid component?
(It isn't added for MIT package though)

**Review by MBilalShafi — APPROVED** [review]
Nice first step to supporting more design systems 🎉

File: packages/grid/x-data-grid-pro/src/material/index.ts (hunk 1 above)
Comment (cherniavskii) [reply packages/grid/x-data-grid-pro/src/material/index.ts:12]: Right, I forgot to include it in the MIT package.
Or is your concern that `ColumnMenu` slot shouldn't be treated as material component?

Comment (MBilalShafi) [reply packages/grid/x-data-grid-pro/src/material/index.ts:13]: Yes, that was my thinking, like other components (Cell, Row, etc), ColumnMenu is not essentially a material component and might be treated as a Grid component.

Would MenuItem and other material slots being used in ColumnMenu be sufficient to support Joy UI? Or you think that we should also consider ColumnMenu to be part of material slots.

Comment (cherniavskii) [reply packages/grid/x-data-grid-pro/src/material/index.ts:13]: Right, `MenuList` and `MenuItem` should be enough for this, and Jun already started the extraction of those slots in https://github.com/mui/mui-x/pull/8059/files#r1119307165 👍

**Comment by MBilalShafi** [comment]
Looks good to me, thanks for the update!

**Comment by cherniavskii** [comment]
I'll wait until https://github.com/mui/mui-x/pull/7968 is merged to avoid causing merge conflicts

**Review by MBilalShafi — APPROVED** [review]

**Pull request merged**